### PR TITLE
add npipe support to docker_swarm_service

### PR DESCRIPTION
**Pull request by Drewster727**
##### SUMMARY
Per the following moby/docker merge that was released with docker 19.03.0:
https://github.com/moby/moby/pull/37400

> SWARM
> - Added support for maximum replicas per node. moby/moby#37940
> - Added support for GMSA CredentialSpecs from Swarmkit configs. moby/moby#38632
> - Added support for sysctl options in services. moby/moby#37701
> - Added support for filtering on node labels. moby/moby#37650
> - **_Windows: Support added for named pipe mounts in docker service create + stack yml. moby/moby#37400_**
> - VXLAN UDP Port configuration now supported. moby/moby#38102
> - Now using Service Placement Constraints in Enforcer. docker/swarmkit#2857
> - Increased max recv gRPC message size for nodes and secrets. docker/engine#256
> 

npipe support has been added for swarm service creation. This should just work as-is, but I did not get a chance to fully test.
This should be as simple as the commit looks, just needed to allow that mount type. It will of course only work with windows.
In my case, I have linux (centos) manager nodes with some windows worker nodes, and I create swarm services thru the linux managers. So, I needed a way to allow npipe creation for the windows nodes.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
docker_swarm_service

##### ADDITIONAL INFO
Here's an example of where this can be applied:
```
- name: Ensure swarm service portainer agent is applied (windows)
  docker_swarm_service:
    name: portainer-agent-win
    image: "portainer/agent:windows-amd64-1.3.0"
    mode: global
    placement:
        constraints:
            - node.platform.os == windows
    restart_config:
      condition: any
      delay: 5s
      max_attempts: 5
      window: 30s
    networks:
        - portainer-agent
    env:
        AGENT_CLUSTER_ADDR: tasks.portainer-agent
    mounts:
        - source: '\\\\.\\pipe\\docker_engine'
          target: '\\\\.\pipe\\docker_engine'
          type: npipe
```

### CODE REVIEW

**Review by felixfontein — APPROVED** [review]
Looks good. You need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to), though. Also, you should document that `npipe` was added in Ansible 2.9.

**Comment by Drewster727** [comment]
@felixfontein I pushed a changelog fragment to my devel branch and it thinks it's already in this PR, but for some reason I don't see that addition here... Is it just slow?

**Comment by felixfontein** [comment]
@Drewster727 interesting, I can see that it is there (https://github.com/Drewster727/ansible/commit/2bd9e71222451c0d26c00381641d628a9df32456) but for some reason GitHub doesn't seem to notice it in this PR... I hope it will just take some time until GitHub updates :)

Anyway, you have to use ReST syntax in the changelog fragment, i.e. something like:
```.yaml
minor_changes:
- docker_swarm_service - Add ``npipe`` mount support.
```

**Comment by felixfontein** [comment]
Funny, now that you added another commit, both commits showed up here...  I guess this is a random GitHub fluke :)

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py [hunk 1]
```diff
@@ -285,12 +285,14 @@
       type:
         description:
           - The mount type.
+          - Note that C(npipe) is only supported by Docker for Windows.
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_swarm_service.py]:
```suggestion
          - Note that C(npipe) is only supported by Docker for Windows. Also note that C(npipe) was added in Ansible 2.9.
```

**Review by felixfontein — APPROVED** [review]
LGTM. @hannseman any comments from your side?

**Comment by hannseman** [comment]
@Drewster727 thanks for this feature PR!

shipit

**Comment by felixfontein** [comment]
@Drewster727 thanks for implementing this!
@hannseman thanks for reviewing!

**Pull request merged**